### PR TITLE
Backend cleanup

### PR DESCRIPTION
**Pull request by lassizci**
Proposal to implement cleanup routine for backends. #554

### CODE REVIEW

File: builtin/logical/postgresql/backend.go [hunk 1]
```diff
@@ -35,6 +35,8 @@ func Backend() *framework.Backend {
 		Secrets: []*framework.Secret{
 			secretCreds(&b),
 		},
+
+		Clean: b.ResetDB,
```
Comment (jefferai) [inline builtin/logical/postgresql/backend.go]: I don't see the actual implementation of ResetDB in the diff...did you mean to include it?

**Comment by jefferai** [comment]
I left one minor comment -- this looks really good!

FYI, some major changes around router/core are going to land soon. I'll poke you when they've landed and you can get this updated -- should be straightforward. Then I'll merge it in.

**Comment by jefferai** [comment]
OK -- those changes landed. Please update and address that one comment!

File: builtin/logical/postgresql/backend.go (hunk 1 above)
Comment (lassizci) [inline builtin/logical/postgresql/backend.go]: ResetDB was there for clean up on configuration change (write to /config) and it made sense to use it as it does exactly what's needed to clean up existing conenctions.

**Comment by lassizci** [comment]
I have rebased the commits and, it should be up to date again.

File: builtin/logical/postgresql/backend.go (hunk 1 above)
Comment (jefferai) [inline builtin/logical/postgresql/backend.go]: That's what I get for looking at diffs in a bubble.

**Comment by jefferai** [comment]
LGTM! Thanks!!!!  :+1:  :ship:

**Pull request merged**